addons/sourcemod/scripting/include/shavit/sql-create-tables-and-migrations.sp

```diff
@@ -540,6 +554,65 @@ public void Trans_DeprecateExactTimeIntFailed(Database db, ArrayStack stack, int
 	LogError("Timer (core) error! ExactTimeInt migration failed. %d %d Reason: %s", numQueries, failIndex, error);
 }
 
+void ApplyMigration_FixSQLiteMapzonesROWID()
+{
+	if (gI_Driver != Driver_sqlite)
+	{
+		InsertMigration(Migration_FixSQLiteMapzonesROWID);
+		return;
+	}
+
+	char sQuery[256];
+	FormatEx(sQuery, sizeof(sQuery), "SELECT EXISTS(SELECT 1 FROM `%smapzones` WHERE id IS NULL);", gS_SQLPrefix);
+
+	QueryLog(gH_SQL, SQL_FixSQLiteMapzonesROWID_Callback, sQuery, 0, DBPrio_High);
+}
+
+public void SQL_FixSQLiteMapzonesROWID_Callback(Database db, DBResultSet results, const char[] error, any data)
+{
+	if (results == null)
+	{
+		LogError("Timer error! SQLiteMapzonesROWID migration failed. Reason: %s", error);
+		return;
+	}
+	else if (!results.FetchInt(0)) // No NULL ids
```

Comment:
This needs `results.FetchRow()` to be called first.
In fact, since `.FetchRow()` returns a bool if a row was fetched, it should be fine to swap it to this:
```suggestion
	else if (!results.FetchRow()) // No NULL ids
```